server/controllers/devices.coffee

```diff
@@ -61,7 +61,7 @@ deviceExists = (login, cb) ->
         else if body.length is 0
             cb null, false
         else
-            cb null, body[0]
+            cb null, body[0]?.value
```

Comment:
It is the source of the bug on travis. `checkLogin` is used for getting the actual doc when updating. Without it, the updated doc lost its `login` field. And when we try to delete the device, it is no longer found, as he has no login.